### PR TITLE
typecheck for Reshape Function

### PR DESCRIPTION
**Pull request by delta2323**
Related to #123

### CODE REVIEW

**Comment by delta2323** [comment]
This PR currently cannot be merged since it needs changes of #204 .

**Comment by beam2d** [comment]
#204 has a new interface that should be deferred to the next minor release. We want to merge this PR before #204, so would you define a local one for prod?

**Comment by delta2323** [comment]
@beam2d I agree with you. For now, I add `prod` to `Reshape` class.

File: chainer/functions/reshape.py [hunk 1]
```diff
@@ -1,13 +1,31 @@
+import numpy
+
 from chainer import function
+from chainer.utils import type_check
 
 
 class Reshape(function.Function):
+    type_check_prod = type_check.Variable(numpy.prod, 'prod')
```
Comment (beam2d) [inline chainer/functions/reshape.py]: Do not write anything above the docstring. How about moving it to module level and name it like `_type_check_prod`?

**Comment by beam2d** [comment]
LGTM except one comment.

**Comment by beam2d** [comment]
[LGTM](http://lgtm.in/i/yPb27nh7q)

**Pull request merged**